### PR TITLE
[codex] Adapt Unicorn Hub control plane

### PR DESCRIPTION
**Pull request by kiaquila**
## Summary

- add `specs/019-unicorn-hub-adaptation` and refresh stale repo docs about the current modular static app
- add `.unicorn-hub/config.json`, shared helpers, and SENAR PR/spec templates adapted to `docs_pallete_maker`
- add event-driven AI Review rerun support through repository-owned policy/rerun scripts and `ai-review-rerun.yml`
- add tests for review markers, rerun routing, and run selection
- patch OSV-reported `fast-uri` transitive advisory through a targeted pnpm override and release-age exception

## SENAR Done Gate

- [x] Feature memory names the goal and scope.
- [x] Every acceptance criterion has evidence in the PR, plan, or linked checks.
- [x] At least one negative scenario is covered or explicitly waived.
- [x] `tasks.md` records dead ends, decisions, and known issues.
- [x] Durable docs are updated when behavior, workflow, deploy, or architecture changes.
- [x] Any remaining known issue is accepted by the human merge owner.

## Validation

- [x] `pnpm run preflight`

### CODE REVIEW

**Comment by kiaquila** [comment]
@codex review

**Comment by kiaquila** [comment]
@codex review

**Comment by kiaquila** [comment]
@codex review

**Comment by kiaquila** [comment]
@codex review

**Comment by kiaquila** [comment]
@codex review

File: scripts/ai-review-rerun.mjs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  }
+
+  const latestCompletedRun = matchingRuns.find(
+    (run) => run.status === "completed",
+  );
+  if (latestCompletedRun?.conclusion === "success") {
+    return { action: "already_success", run: latestCompletedRun };
+  }
+  if (rerunnableConclusions.has(latestCompletedRun?.conclusion)) {
+    return { action: "rerun", run: latestCompletedRun };
```
[the remark this answers is not in the document]
Comment (kiaquila) [reply scripts/ai-review-rerun.mjs:111]: Addressed across commits 9138348 and 36269c5.

`selectAiReviewRun` (scripts/ai-review-rerun.mjs:74-109) now sorts runs by `created_at` desc, evaluates `activeStatuses` first, then compares `evidenceCreatedAt` against `latestCompletedRun.created_at`: when trusted review evidence is newer than the latest green run, the function returns `rerun` before the `already_success` branch (lines 87-99). Covered by tests/ai-review-rerun.test.mjs:240-258 ("reruns a successful run when newer review evidence arrives").

The follow-up commit 36269c5 closes a related edge case: `evidenceTimestamp` now ignores `comment.updated_at` so trusted bot comment edits cannot spam fresh reruns (tests/ai-review-rerun.test.mjs:260-278), and the inline `rerunAiReviewForPrHead` call in scripts/ai-command-policy.mjs has been removed so that .github/workflows/ai-review-rerun.yml is the single source of truth for evidence-driven reruns.

File: scripts/ai-review-rerun.mjs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  if (activeRun) {
+    return { action: "already_running", run: activeRun };
+  }
+
+  const rerunnableRun = matchingRuns.find(
+    (run) =>
+      run.status === "completed" && rerunnableConclusions.has(run.conclusion),
+  );
+  if (rerunnableRun) {
+    return { action: "rerun", run: rerunnableRun };
```
[the remark this answers is not in the document]
Comment (kiaquila) [reply scripts/ai-review-rerun.mjs]: Addressed by commit 9138348 and reaffirmed by 36269c5.

`selectAiReviewRun` (scripts/ai-review-rerun.mjs:74-109) sorts `matchingRuns` by `created_at` desc, so `latestCompletedRun` is always the most recent completed run. When that run is a success and no newer trusted evidence has arrived, the function short-circuits to `already_success` (lines 101-103) before considering the rerunnable-failure branch. Older failed runs are never selected over a newer success. Covered by tests/ai-review-rerun.test.mjs:213-238 ("does not rerun an older failure after a newer success").

[the remark this answers is not in the document]
Comment (kiaquila) [reply scripts/ai-review-rerun.mjs]: Addressed by commit 9138348.

`selectAiReviewRun` (scripts/ai-review-rerun.mjs:101-103) returns `already_success` for the latest completed run when its conclusion is "success" and no newer trusted review evidence is present (the evidence-newer check at lines 87-99 runs before this branch). Older failed runs cannot displace a newer pass. Covered by the same regression test above (tests/ai-review-rerun.test.mjs:213-238).

**Comment by kiaquila** [comment]
@codex review

HEAD now `36269c5`. Iteration addresses local reviewer feedback on top of the P1/P2 thread fixes:
- architect HIGH: removed inline `rerunAiReviewForPrHead` from `scripts/ai-command-policy.mjs` so `.github/workflows/ai-review-rerun.yml` is the single source of truth for evidence-driven reruns
- code-reviewer HIGH: `evidenceTimestamp` in `scripts/ai-review-rerun.mjs` now ignores `comment.updated_at` so trusted bot edits cannot spam reruns (test at tests/ai-review-rerun.test.mjs:260-278)
- security MEDIUM: `trustedReviewLoginsForAgent` in `scripts/ai-review-helpers.mjs` now validates user-supplied logins against `^[a-z0-9-]+\[bot\]$` and drops non-bot entries (test at tests/ai-review-rerun.test.mjs:110-127)
- cleanup: removed dead `trustedReviewLogins*` placeholder keys from `.unicorn-hub/config.json`

`pnpm run preflight` passes locally (69/69 tests).

File: scripts/ai-review-rerun.mjs [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  const activeRun = matchingRuns.find((run) => activeStatuses.has(run.status));
+  if (activeRun) {
+    return { action: "already_running", run: activeRun };
+  }
+
+  const evidenceTime = Date.parse(evidenceCreatedAt || "");
+  const latestCompletedRun = matchingRuns.find(
+    (run) => run.status === "completed",
+  );
+  const latestCompletedTime = Date.parse(latestCompletedRun?.created_at || "");
```
[the remark this answers is not in the document]
Comment (kiaquila) [reply scripts/ai-review-rerun.mjs]: Addressed in commit 971114e.

`selectAiReviewRun` (scripts/ai-review-rerun.mjs:74-115) now selects `latestCompletedRun` and sorts `matchingRuns` via `latestRunActivityTime`, which reads `run.updated_at` first and falls back to `run.run_started_at` and `run.created_at`. `workflow_run.updated_at` advances when each rerun attempt completes, so once a rerun for evidence comment T1 finishes at T2, that same comment stops looking newer than the latest pass (T1 < T2 → `already_success`). Regression test at tests/ai-review-rerun.test.mjs:260-281 ("treats `updated_at` as the latest run activity to suppress post-rerun edit storms") simulates the edited-evidence-after-rerun scenario.

**Comment by kiaquila** [comment]
@codex review

HEAD now `971114e`. Addresses the P2 you raised on `36269c5`:
- `workflow_run.created_at` is immutable across rerun attempts, so comparing evidence timestamps against it always looked "newer" once any evidence post-dated the original AI Review run
- `selectAiReviewRun` now reads run freshness from `updated_at` first (with fallback to `run_started_at` and then `created_at`); once a rerun attempt completes, the same evidence comment stops triggering more reruns
- Regression test at tests/ai-review-rerun.test.mjs:260-281

`pnpm run preflight` passes locally (70/70 tests).
